### DISCUSSION TITLE
Running specific Cypress test cases which are not in common folder and also if they are huge for parsing in cli

### DISCUSSION THREAD

**Discussion by SwardCoder** [cypress-io/cypress#24066] [Questions and Help]
Hi Team 
is there any way I want to pass the paths of the test cases which I wish to run in a file or dictionary 
and run them via cypress CLI

Example 
File1.txt
-- abc/cd/Doc1.spec.ts
-- xyz/sa/Test.spec.ts
-- cvd/dc/Testing.spec.ts

and I want to run all the specs which are in the text file  via CLI 

I don't want to make CLI manually like 
npx cypress run --spec="abc/cd/Doc1.spec.ts, xyz/sa/Test.spec.ts ..........."
because if there are 1000 test cases failing then its a problem for me with specific strings if I wish to use the above approach 

so please help me with your valuable ideas 
thank you 🙂 

the intention behind this is if I get some 3 test cases failing I will maintain them in a file 
then I will ask developers to fix the appropriate things then once they fix it 
I would like to run only this first 
Note: I would consider failing of these test cases even after using the retries option for flaky test cases

**Comment by portyu9** [comment]
Hi @SwardCoder 

Cypress does not currently have a CLI option such as:

```bash
cypress run --spec-file failed-specs.txt
```

For a large list, you can read the file from `cypress.config.js` and use the paths as `specPattern`:

```js
const { defineConfig } = require('cypress')
const fs = require('fs')

const failedSpecs = fs
  .readFileSync('failed-specs.txt', 'utf8')
  .split(/\r?\n/)
  .map((s) => s.trim())
  .filter(Boolean)

module.exports = defineConfig({
  e2e: {
    specPattern: failedSpecs,
  },
})
```

Then your file can simply contain:

```text
abc/cd/Doc1.cy.ts
xyz/sa/Test.cy.ts
cvd/dc/Testing.cy.ts
```

and run:

```bash
npx cypress run
```

This avoids building a huge `--spec="a,b,c,..."` command entirely.

One distinction: this selects **spec files**, not individual `it()` test cases inside those files. If your failure list contains individual tests rather than spec paths, you would need test filtering instead.

Please mark this answer as accepted if it helped.
